js/api.md

```diff
@@ -461,8 +461,8 @@ Also, you can use `subscription` and `onSubscriptionMsg` attributes to enable su
           onSubscriptionMsg={(prev, { subscribeToEventComments }) => {
             console.log ( subscribeToEventComments);
             return prev; }}>
-    {({ data: { listEvents } }) => (
-        <AllEvents events={listEvents ? listEvents.items : []} />
+    {({ data: { listEvents }, loading }) => (
```

Comment:
Similar to the above, let's be more prescriptive on the error handling and also call out the function argument from the GraphQL response. Here is a suggestion for the code and text:

```
        <Connect query={graphqlOperation(ListTodos)}
          subscription={graphqlOperation(OnCreateTodo)}
          onSubscriptionMsg={(prev, {onCreateTodo}) => {
              console.log('Subscription data:', onCreateTodo)
              return prev;
            }
          }>
        {({ data: { listTodos }, loading, error }) => {
          if (error) return <h1>Error</h1>;
          if (loading || !listTodos) return <h1>Loading...</h1>;
            return <ListView todos={listTodos.items} />
        }}
        </Connect>
```

Note - In the above example `OnCreateTodo` passed to the `subscription` prop as `subscription={graphqlOperation(OnCreateTodo)}` is the name of your GraphQL document. For example when used with the Amplify CLI it will generate a document and you would use `import { OnCreateTodo } from './graphql/subscriptions';` at the top of your app code. **However** the argument `onCreateTodo` received by the `onSubscriptionMsg` is defined in the GraphQL selection set. For example your GraphQL statement would look like the following here:

```
export const OnCreateTodo = `
  subscription OnCreateTodo {
    onCreateTodo {
      id
      name
      description
    }
  }
`;
```